clustered/client/src/main/java/org/ehcache/clustered/client/config/Timeouts.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.clustered.client.config;
+
+import org.ehcache.clustered.client.internal.ClusterTierManagerClientEntity;
+
+import java.time.Duration;
+import java.util.Objects;
+
+import java.util.function.LongSupplier;
+
+/**
+ * Describes the timeouts for {@link ClusterTierManagerClientEntity} operations.  Use
+ * {@link #builder()} to construct an instance.
+ */
+public final class Timeouts {
+
+  public static final Duration DEFAULT_OPERATION_TIMEOUT = Duration.ofSeconds(5);
+  public static final Duration INFINITE_TIMEOUT = Duration.ofMillis(Long.MAX_VALUE);
+  public static final Timeouts DEFAULT = Timeouts.builder().build();
+
+  private final Duration readOperationTimeout;
+  private final Duration writeOperationTimeout;
+  private final Duration connectionTimeout;
+
+  private Timeouts(Duration readOperationTimeout, Duration writeOperationTimeout, Duration connectionTimeout) {
+    this.readOperationTimeout = readOperationTimeout;
+    this.writeOperationTimeout = writeOperationTimeout;
+    this.connectionTimeout = connectionTimeout;
+  }
+
+  public Duration getReadOperationTimeout() {
+    return readOperationTimeout;
+  }
+
+  public Duration getWriteOperationTimeout() {
+    return writeOperationTimeout;
+  }
+
+  public Duration getConnectionTimeout() {
+    return connectionTimeout;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+
+    Timeouts timeouts = (Timeouts) o;
+
+    if (!readOperationTimeout.equals(timeouts.readOperationTimeout)) {
+      return false;
+    }
+    if (!writeOperationTimeout.equals(timeouts.writeOperationTimeout)) {
+      return false;
+    }
+    return connectionTimeout.equals(timeouts.connectionTimeout);
+  }
+
+  @Override
+  public int hashCode() {
+    int result = readOperationTimeout.hashCode();
+    result = 31 * result + writeOperationTimeout.hashCode();
+    result = 31 * result + connectionTimeout.hashCode();
+    return result;
+  }
+
+  public static Builder builder() {
```

Comment:
This is not consistent with the way we have builders as top level classes in a different package. I would instead move the builder to `<package>.builders.TimeoutsBuilder` for consistency.

Reply:
Like `Expirations` :-) I thought about it and kinda let it slip as an exception. But I can

Reply:
On problem with that is that the constructor can't be private anymore. Before, we were forcing using the builder